token-metadata/program/tests/utils/digital_asset.rs

```diff
@@ -1262,9 +1263,19 @@ impl DigitalAsset {
             .get_account(self.edition.unwrap())
             .await?;
 
-        assert!(md_account.is_none());
+        // Token Metadata accounts may still be open because they are no longer being re-assigned
+        // to the system program immediately, but if they exist they should have a
+        // data length of 0.
+
+        if let Some(account) = md_account {
+            assert_eq!(account.data.len(), 0);
+        }
```

Comment:
There are two situations here: one where the Metadata account is still open either because it has not been garbage collected yet, or because it still has fees on it. In that case we use the `if let` to see if it has an account data length of 0. In the case where it's been garbage collected it's the `None` case and the else is implicit. The only time to fail is if it exists and the account length isn't 0.

Reply:
Ok that is fine.  I thought all the nonfungible accounts used in tests would now have uncollected fees on them from `create` and remain open since the tests don't remove fees.